### PR TITLE
Proper cost function

### PR DESCRIPTION
**Pull request by jesper-friis**
This PR builds on top of PR #67 

It makes the cost function independent of any data, such that the costs can be calculated purely from the triplestore (from which data may be accessed). This is needed by OntoFlow.

### CODE REVIEW

File: tripper/mappings/mappings.py [hunk 1]
```diff
@@ -318,9 +323,13 @@ def lowest_costs(self, nresults: int = 5) -> "List[Tuple[float, int]]":
             # mapping step.
             if callable(self.cost):
                 for i, rno in enumerate(base.routeno):
-                    values = get_values(inputs, rno, magnitudes=True)
-                    owncost = self.cost(**values)
+                    inputs, _ = self.get_inputs(rno)
+                    input_iris = [input.output_iri for input in inputs.values()]
+                    owncost = self.cost(self.triplestore, input_iris, self.output_iri)
                     base.cost[i] += owncost
+                    # values = get_values(inputs, rno, magnitudes=True)
```
Comment (quaat) [inline tripper/mappings/mappings.py:330]: Delete commented-out code

File: tripper/mappings/mappings.py [hunk 2]
```diff
@@ -588,7 +597,10 @@ def getcost(target, stepname):
         cost = soCost.get(target, default_costs[stepname])
         if cost is None:
             return None
-        return function_repo[cost] if cost in function_repo else float(cost)
+        # return function_repo[cost] if cost in function_repo else float(cost)
```
Comment (quaat) [inline tripper/mappings/mappings.py:600]: Delete commented out code

**Review by quaat — CHANGES_REQUESTED** [review]
Looks ok. Just remove commented out code and merge

File: tripper/mappings/mappings.py (hunk 1 above)
Comment (jesper-friis) [reply tripper/mappings/mappings.py:330]: done

**Pull request merged**